### PR TITLE
add libmsquic to one helix image

### PR DESCRIPTION
**Pull request by wfurt**
For now, updating Fedora only to verify that we can run QUIC tests at least on one Linux distro.
There may be better way to establish trust but I did not see external files so I inline everything to the Dockerfile. 

cc: @safern @ManickaP @MattGal

### CODE REVIEW

File: src/fedora/32/helix/amd64/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,36 @@
 FROM mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-32
 
+# Add MS package repo
+RUN dnf config-manager --add-repo https://packages.microsoft.com/fedora/32/prod/ && \
```
Comment (ManickaP) [inline src/fedora/32/helix/amd64/Dockerfile:4]: Can't we cURL/wget the GPP public key?
Or are we constraint here by something or is it frown upon to not to hard code?

**Review by ManickaP — COMMENTED** [review]
Also shouldn't the package go rather to the machine where tests run? Having msquic present during runtime build is not a requirement, but having it present when running test is.
I don't fully understand the image hierarchy here, so maybe this will get wherever it needs.

Finally, are we doing only fedora? Or is this just "test the waters" PR and than we'll add it to other distros as well?

File: src/fedora/32/helix/amd64/Dockerfile (hunk 1 above)
Comment (MattGal) [reply src/fedora/32/helix/amd64/Dockerfile:4]: No reason to not just curl this from a well-known public location if possible, lots of locations for it on the internet.

**Comment by MattGal** [comment]
> Also shouldn't the package go rather to the machine where tests run? Or does the "helix" in path means exactly that? Having msquic present during runtime build is not a requirement, but having it present when running test is.
> I don't fully understand the image hierarchy here, so maybe this will get wherever it needs.

The repo name predates the existence of Helix; it was originally used only for making build images, now it's used for several types of images.  The ones with helix in the path, e.g. `src/fedora/32/helix/amd64/Dockerfile` are indeed only intended for test execution.  Merging PRs here pushes the images to the Microsoft Container Registry, which can then be used in test work items by specifying `<Helix Queue Name>@<fully qualified MCR docker tag>`. 

As for only doing it to one image, I dunno; perhaps the goal is to make sure this is a viable approach before putting the change in every image.

File: src/fedora/32/helix/amd64/Dockerfile (hunk 1 above)
Comment (wfurt) [reply src/fedora/32/helix/amd64/Dockerfile:4]: we could. But I did not want to establish trust by depending on internet. If we want to do that, I think we should hardcore sha and verify that what ever we download is exactly what we expect. I would be probably ok with gpgp registry but I did not find our keys there.

**Comment by wfurt** [comment]
And yes, this is the machine - at least for runs where we run tests in containers. We will need to make follow-up change in runtime once the image is built. 

And also yes, one image as canary. When we get good results we can spread the goodness (quicness) to more. I don't have good plan for queue running full VMs. I started with container because it is easy.

**Review by MattGal — APPROVED** [review]
I'm signing off so you can try this out, but I think the "right thing to do" is to put microsoft.asc into the repo and use [docker cp](https://docs.docker.com/engine/reference/commandline/cp/) to bring it in, it makes a much prettier dockerfile.

**Pull request merged**